src/jet/elements/environment.rs

```diff
@@ -100,6 +100,16 @@ impl ElementsEnv {
         &self.c_tx_env
     }
 
+    /// Returns the transaction of this environment
+    pub fn tx(&self) -> &Arc<Transaction> {
+        &self.tx
+    }
```

Comment:
In 8365c621bbc93d1c82f38ce53c1f9073d90ac339:

You should return a `&Transaction` rather than leaking the internal detail that you have an `Arc`. You could add a separate accessor to get a clone of the `Arc` if you want.

TBH I'm not sure why we have an `Arc` here at all. Why can't the `ElementsEnv` directly own its `Transaction`? Or use a normal reference? You could maybe parameterize it over `Deref<Target = Transaction>` like we do with `SighashCache`.

Reply:
I don't see any uses of `tx` in the library that would not be possible with a reference. Maybe the `Arc` is a "historic artifact". @sanket1729 Any ideas?

Reply:
There is a big comment here on why it was needed: https://github.com/BlockstreamResearch/rust-simplicity/blob/a57e6a4f32f0ef04839329621bed861ce09bf076/src/jet/elements/environment.rs#L45-L53

Reply:
But we don't have extension trait anymore. I think can be cleaned up to use &'a or AsRef.